src/java.desktop/share/classes/javax/swing/plaf/basic/BasicDirectoryModel.java

```diff
@@ -360,12 +360,15 @@ public DoChangeContents call() {
                                 break;
                             }
                         }
-                        if (start >= 0 && end > start
-                            && newFileCache.subList(end, newSize).equals(fileCache.subList(start, oldSize))) {
-                            if (loadThread.isInterrupted()) {
-                                return null;
+
+                        if (start >= 0 && end > start) {
+                            List<File> listStart_OldSize = new Vector<>(fileCache.subList(start, oldSize));
```

Comment:
Why Vector was used instead of ArrayList ?

Reply:
Nothing specific reason why I choose vector apart from the main object is of Vector type and since in this context (where both the objects are not shared) both Vector/Arraylist comparison remains same. Does Arraylist comparison has any advantage here?

Reply:
I already looked at this and Vector is what this class uses already and there's no reason to be different and definitely no reason to change the rest.